xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -148,7 +149,15 @@ void adjustResourceSubscription(XdsResourceType<?> resourceType) {
     }
     if (adsStream == null) {
       startRpcStream();
+      // when the stream becomes ready, it will send the discovery requests
+      return;
+    }
+
+    // We will do the rest of the method as part of the readyHandler when the stream is ready.
+    if (!isConnected()) {
```

Comment:
Having a `isInBackoff()` check and `isConnected()` check is confusing, as it is hard to think about which is appropriate. It appears that most usages of `isInBackoff()` in this class should be `isConnected()` now that you've changed when the requests are queued. Outside of this class, have all usages of `isInBackoff()` deleted? (Probably not in tests)